### PR TITLE
misc: lint xdsl

### PR DESCRIPTION
**Pull request by superlopuh**
This finishes my procrastilinting journey to 0 problems detected in VSCode. We have gained a new feature: whenever we change an API, the IDE will show us exactly the areas to look at, and not in a soup of irrelevant errors. My hope is that this will increase everyone's productivity and well-being.

Screenshot 2023-03-17 at 11 32 09

The changes in this PR include:

1. removing the difference between the CI pyright and the local one, they are now in sync
2. linting every file in the repository by default, not just in the `xdsl` folder
3. Moving all the files with remaining errors, such as tests, into an `exclude` list, to be linted later
    - the errors in the file will be shown only if the file is open
    - people working on those files can gradually get those errors to 0, and remove them from the exclude list
4. adding an "ignore" list of files that will not show errors even if we navigate to them
    - this includes the previously excluded irdl printer, I don't have it in me to lint that file, and it seems non-critical
    - this also includes the filecheck files, and some more utility files that we probably don't want to lint at all

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,7 +5,37 @@ requires = [
...
+    "tests/dialects/test_mpi_lowering.py",
+    "tests/dialects/test_vector.py",
+    "tests/xdsl_opt/test_xdsl_opt.py",
+]
+"ignore" = [
+    "venv",
+    "versioneer.py",
+    "setup.py",
+    "update_xdsl_pyodide_build.py",
+    "xdsl/irdl_mlir_printer.py",
```
Comment (superlopuh) [inline pyproject.toml:37]: this might be worth a task for someone who wants to contribute to the project?

Comment (superlopuh) [reply pyproject.toml:37]: I no longer have it in me

File: xdsl/dialects/builtin.py [hunk 2]
```diff
@@ -45,8 +45,7 @@ def verify(self, attr: Attribute) -> None:
 class ArrayAttr(GenericData[tuple[AttributeCovT, ...]]):
     name: str = "array"
 
-    def __init__(self: ArrayAttr[AttributeCovT],
```
Comment (superlopuh) [inline xdsl/dialects/builtin.py:48]: For some reason this confuses the linter

File: .github/workflows/update_xdsl_pyodide_build.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 # This script updates the meta.yaml file used by Pyodide to bundle and use xDSL
 # Takes the .yaml file and the xDSL directory as arguments
 
-import yaml
+import yaml  # type: ignore
```
[the remark this answers is not in the document]
Comment (superlopuh) [reply .github/workflows/update_xdsl_pyodide_build.py:6]: For some reason this is a super unignorable error, but of a VSCode variety as opposed to pyright

File: pyproject.toml (hunk 1 above)
Comment (math-fehr) [reply pyproject.toml:37]: What's the difference between `ignore` and `exclude` here?

**Review by math-fehr — APPROVED** [review]
:confetti_ball:

**Comment by superlopuh** [comment]
ignore means that even if there are errors/warnings they won't be shown in VSCode

**Pull request merged**